File: String/2315.Count Asterisks/Solution.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        ans = cur = 0
        mid = False
        for c in s:
            if c == '*':
                cur += 1
            if c == '|':
                if not mid:
                    ans += cur
                mid = not mid
                cur = 0
        return ans + cur

File: String/2315.Count Asterisks/Solution.java
class Solution {
    public int countAsterisks(String s) {
        int ans = 0, cur = 0;
        boolean mid = false;
        for (char c : s.toCharArray()) {
            if (c == '*') {
                cur++;
            }
            if (c == '|') {
                if (!mid) {
                    ans += cur;
                }
                mid = !mid;
                cur = 0;
            }
        }
        return ans + cur;
    }
}